adapters/oidc/adapter-core/src/main/java/org/keycloak/adapters/authorization/KeycloakAdapterPolicyEnforcer.java

```diff
@@ -72,7 +79,27 @@ protected boolean isAuthorized(PathConfig pathConfig, PolicyEnforcerConfig.Metho
         AccessToken.Authorization newAuthorization = accessToken.getAuthorization();
 
         if (newAuthorization != null) {
-            authorization.getPermissions().addAll(newAuthorization.getPermissions());
+            List<Permission> grantedPermissions = authorization.getPermissions();
```

Comment:
What do you think about using Set<Permission> here? We would avoid this merging, but it would be maybe to complicated change as we would need to change it also in AccessToken.Authorization and I don't know whether it is possible.

Reply:
Yeah, the thing is that this change impacts backward compatibility and I prefer to not mess with that right now as we have a plenty of other changes to APIs.

Reply:
Yeah I thought so, ok then no problem.